nuxeo-mongodb-audit/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditBackend.java

```diff
@@ -0,0 +1,358 @@
+
+    public static final String SEQ_NAME = "audit";
+
+    protected MongoCollection<Document> collection;
+
+    protected MongoDBLogEntryProvider provider = new MongoDBLogEntryProvider();
+
+    public MongoDBAuditBackend(NXAuditEventsService component, AuditBackendDescriptor config) {
+        super(component, config);
+    }
+
+    @Override
+    public int getApplicationStartedOrder() {
+        DefaultComponent component = (DefaultComponent) Framework.getRuntime().getComponent(MongoDBComponent.NAME);
+        return component.getApplicationStartedOrder() + 1;
+    }
+
+    @Override
+    public void onApplicationStarted() {
+        log.info("Activate MongoDB backend for Audit");
+        // First retrieve the collection name
+        ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+        String collName = configurationService.getProperty(COLLECTION_NAME_PROPERTY, DEFAULT_COLLECTION_NAME);
+        // Get a connection to MongoDB
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        // Get database
+        MongoDatabase database = mongoService.getDatabase(AUDIT_DATABASE_ID);
+        // Get collection
+        collection = database.getCollection(collName);
+        // TODO migration ?
+    }
+
+    @Override
+    public void onShutdown() {
+        if (collection != null) {
+            collection = null;
+        }
+    }
+
+    /**
+     * @return the {@link MongoCollection} configured with audit settings.
+     */
+    public MongoCollection<Document> getAuditCollection() {
+        return collection;
+    }
+
+    @Override
+    public List<LogEntry> getLogEntriesFor(String uuid, String repositoryId) {
+        Bson docFilter = Filters.eq(PROPERTY_DOC_UUID, uuid);
+        Bson repoFilter = Filters.eq(PROPERTY_REPOSITORY_ID, repositoryId);
+        Bson query = Filters.and(docFilter, repoFilter);
+        return getLogEntries(query, false);
+    }
+
+    @Override
+    public List<LogEntry> getLogEntriesFor(String uuid, Map<String, FilterMapEntry> filterMap, boolean doDefaultSort) {
+        Bson docFilter = Filters.eq(PROPERTY_DOC_UUID, uuid);
+
+        Bson filter;
+        if (MapUtils.isEmpty(filterMap)) {
+            filter = docFilter;
+        } else {
+            List<Bson> list = new ArrayList<>(Collections.singleton(docFilter));
+            for (Entry<String, FilterMapEntry> entry : filterMap.entrySet()) {
+                FilterMapEntry filterEntry = entry.getValue();
+                list.add(Filters.eq(filterEntry.getColumnName(), filterEntry.getObject()));
+            }
+            filter = Filters.and(list);
+        }
+        return getLogEntries(filter, doDefaultSort);
+    }
+
+    @Override
+    public LogEntry getLogEntryByID(long id) {
+        Document document = collection.find(Filters.eq(MongoDBSerializationHelper.MONGODB_ID, Long.valueOf(id)))
+                                      .first();
```

Comment:
collection.findOne() doesn't exist in new MongoDB 3.0 API. Here we use a MongoCollection<Document> and not a DBCollection which was deprecated.